src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Status.all);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const handleCloseModal = () => {
+    setSelectedTodo(null);
+  };
+
+  const filterTodosByStatus = (statusTodos: Todo[], status: Status) => {
+    switch (status) {
+      case Status.all:
+        return statusTodos;
+      case Status.completed:
+        return statusTodos.filter(todo => todo.completed);
+      case Status.active:
+        return statusTodos.filter(todo => !todo.completed);
+      default:
+        return statusTodos;
+    }
+  };
+
+  const filterTodosByQuery = (queryTodos: Todo[], givenQuery: string) => {
+    return queryTodos.filter(todo => todo.title.toLowerCase().includes(givenQuery.toLowerCase()));
+  };
+
+  const displayedTodos = useMemo(() => {
+    const todosByStatus = filterTodosByStatus(todos, filter);
+    const todosByQuery = filterTodosByQuery(todosByStatus, query);
+
+    return todosByQuery;
+  }, [todos, filter, query]);
```

Comment:
you can putt all of that inside one useMemo, there is no need for creating two separate functions here.